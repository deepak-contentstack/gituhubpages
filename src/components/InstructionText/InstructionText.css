@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/common.css';

.InstructionText {
    color: $color-font-secondary;
    display: block;
    font-family: $font-family-primary;
    font-size: $size-font-small;
    line-height: $line-height-default;
    margin: calc(var(--space-2) + 1px) 0 0 $space-10;
}