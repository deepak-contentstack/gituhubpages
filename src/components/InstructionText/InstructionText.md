# InstructionText

## Best practices

- Use direct, succinct text that helps the user to successfully complete the form.
- Do not use punctuation.
- Use sentence style in which only the first word of a sentence or term is capitalized.