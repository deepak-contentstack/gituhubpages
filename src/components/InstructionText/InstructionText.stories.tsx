import React from 'react';
import notes from './InstructionText.md';
import InstructionText from './InstructionText';
import DocSubData from '../../utils/components/DocSubData';

const componentDescription = 'The InstructionText component lets you pass on information to the user about a form component.'

const usageDescription = 'This component lets you inform a user more about specific details for a particular field in your form. You can use it only in the context of a form as a corresponding guidance to an input field.'

export default {
  title: 'Components / InstructionText',
  component: InstructionText,
  argTypes: {
    label: {
      control: 'text'
    },
    className: {
      control: 'disabled'
    },
    style: {
      control: 'disabled'
    },
    children: {
      control: 'disabled'
    }
  },
  parameters: {
    componentSubtitle: <DocSubData componentDescription={componentDescription} usageDescription={usageDescription} />,
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/QHcXcnGhGDBQ49J1kRwzYf/02-%E2%80%A2-Molecules?node-id=233%3A5666',
    },
    notes: { notes }
  }
}

export const Default = (args) => <InstructionText>{args.label}</InstructionText>
Default.args = {
  label: 'Instruction text here'
}


