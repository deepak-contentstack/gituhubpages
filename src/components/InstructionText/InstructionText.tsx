import React, { Component } from 'react';
import cn from 'classnames';
import './InstructionText.css';

export type InstructionTextProps = {
    /**
    * Pass the class names to be appended to this prop
    */
    className?: string;
    /**
    * Pass an ID that you can use for testing purposes. It is applied as a data attribute (data-test-id)
    */
    testId?: string;
    /**
    * Add the styles that can be added directly to the component  
    */
    style?: React.CSSProperties;
    children?: React.ReactNode;
} & typeof defaultProps;

const defaultProps = { testId: 'cs-instruction-text' };

export class InstructionText extends Component<InstructionTextProps> {
    static defaultProps = defaultProps;

    render() {
        const { className, children, testId, ...otherProps } = this.props;

        const classNames = cn('InstructionText', className);

        return (
            <p {...otherProps} className={classNames} data-test-id={testId}>
                {children}
            </p>
        );
    }
}

export default InstructionText;