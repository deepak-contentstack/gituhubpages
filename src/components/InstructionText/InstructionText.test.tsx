import React from 'react';
import { storiesOf } from '@storybook/react';
import renderer from 'react-test-renderer'
import InstructionText from './InstructionText';

describe('InstructionText', () => {
    it('render default InstructionText', () => {
        const DefaultInstructionText = () => {
            return (
                <InstructionText>InstructionText text here</InstructionText>
            )
        }
        const tree = renderer.create(<DefaultInstructionText />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('renders the component with an additional class name', () => {
        const DefaultInstructionText = () => {
            return (
                <InstructionText className="instruction-text">InstructionText text here</InstructionText>
            )
        }
        const tree = renderer.create(<DefaultInstructionText />).toJSON();
        expect(tree.props.className.includes("instruction-text")).toBe(true);
    })

    it('renders the component with an additional style', () => {
        const DefaultInstructionText = () => {
            return (
                <InstructionText style={{ color: 'red' }}>InstructionText text here</InstructionText>
            )
        }
        const tree = renderer.create(<DefaultInstructionText />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})