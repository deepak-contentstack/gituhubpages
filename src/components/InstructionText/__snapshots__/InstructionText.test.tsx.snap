// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionText render default InstructionText 1`] = `
<p
  className="InstructionText"
  data-test-id="cs-instruction-text"
>
  InstructionText text here
</p>
`;

exports[`InstructionText renders the component with an additional style 1`] = `
<p
  className="InstructionText"
  data-test-id="cs-instruction-text"
  style={
    Object {
      "color": "red",
    }
  }
>
  InstructionText text here
</p>
`;
